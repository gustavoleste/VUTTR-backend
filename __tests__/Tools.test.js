const { Tools, connectDatabase } = require("../src/database/index");
const { databaseURL } = require("../src/config");
const { singleTool } = require("../src/helpers/index");
require("dotenv").config();

describe("Tools Model", () => {
  beforeAll(async () => {
    await connectDatabase(databaseURL, "toolsModelTest");
  });

  afterAll(async () => {
    const database = await connectDatabase(databaseURL, "toolsModelTest");
    await database.dropDatabase();
  });

  it("should not create a empty tool", async () => {
    try {
      const newTool = new Tools({});
      await newTool.save();
    } catch (error) {
      expect(error).toMatchSnapshot();
    }
  });

  it("should not create a tool without the title field set", async () => {
    try {
      const toolWithoutTitle = { ...singleTool, title };
      const newTool = new Tools(toolWithoutTitle);
      await newTool.save();
    } catch (error) {
      expect(error).toMatchSnapshot();
    }
  });

  it("should not create a tool without the link field set", async () => {
    try {
      const toolWithoutLink = { ...singleTool, link };
      const newTool = new Tools(toolWithoutLink);
      await newTool.save();
    } catch (error) {
      expect(error).toMatchSnapshot();
    }
  });

  it("should not create a tool without the tags field set", async () => {
    try {
      const toolWithoutTags = { ...singleTool, tags };
      const newTool = new Tools(toolWithoutTags);
      await newTool.save();
    } catch (error) {
      expect(error).toMatchSnapshot();
    }
  });

  it("should not create a tool without the description field set", async () => {
    try {
      const toolWithoutDescription = { ...singleTool, description };
      const newTool = new Tools(toolWithoutDescription);
      await newTool.save();
    } catch (error) {
      expect(error).toMatchSnapshot();
    }
  });

  it("should create a new tool", async () => {
    try {
      const newTool = new Tools(singleTool);
      await newTool.save();
      expect(newTool).toMatchSnapshot();
    } catch (error) {
      console.log(error);
    }
  });
});
