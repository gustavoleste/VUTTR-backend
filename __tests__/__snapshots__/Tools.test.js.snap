// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tools Model should create a new tool 1`] = `
Object {
  "__v": 0,
  "_id": "4ceda7b37085d444ec1bec64",
  "createdAt": 2019-05-28T21:28:37.290Z,
  "description": "Local app manager. Start apps within your browser, developer tool with local .localhost domain and https out of the box.",
  "link": "https://github.com/typicode/hotel",
  "tags": Array [
    "node",
    "organizing",
    "webapps",
    "domain",
    "developer",
    "https",
    "proxy",
  ],
  "title": "hotel",
  "updatedAt": 2019-05-28T21:28:37.290Z,
}
`;

exports[`Tools Model should not create a empty tool 1`] = `[ValidationError: Tools validation failed: description: Path \`description\` is required., link: Path \`link\` is required., title: Path \`title\` is required.]`;

exports[`Tools Model should not create a tool without the description field set 1`] = `[ReferenceError: description is not defined]`;

exports[`Tools Model should not create a tool without the link field set 1`] = `[ReferenceError: link is not defined]`;

exports[`Tools Model should not create a tool without the tags field set 1`] = `[ReferenceError: tags is not defined]`;

exports[`Tools Model should not create a tool without the title field set 1`] = `[ReferenceError: title is not defined]`;
